tmt/steps/discover/shell.py

```diff
@@ -105,31 +108,55 @@ def _normalize_require(
 
     # Our own implementation, parent uses `name` and `how`, and tests don't have any `how`.
     @classmethod
-    def from_raw(cls: Type[T], raw_data: Dict[str, Any], logger: tmt.utils.Common) -> T:
-        """
-        Unserialize step data instance from its a raw representation.
-        """
+    def from_spec(  # type: ignore[override]
+            cls: Type[T],
+            raw_data: Dict[str, Any],
+            logger: tmt.utils.Common
+            ) -> T:
+        """ Convert to a form suitable for saving in a specification file or CLI option """
 
         data = cls(name=raw_data['name'], test=raw_data['test'])
         data._load_keys(raw_data, cls.__name__, logger)
 
         return data
 
+    def to_spec(self) -> Dict[str, Any]:
+        """ Convert from a specification a file or CLI option """
+
+        data = super().to_spec()
+        data['link'] = self.link.to_spec() if self.link else None
+        data['require'] = [
+            require if isinstance(require, str) else require.to_spec()
+            for require in self.require
+            ]
+
+        return data
+
     def to_serialized(self) -> Dict[str, Any]:
-        """
-        Return keys and values in the form allowing later reconstruction.
+        """ Convert to a form suitable for saving in a file """
 
-        Used to transform container into a structure one can save in a
-        YAML file, and restore it later.
+        data = super().to_serialized()
 
-        See :py:meth:`from_serialized` for its counterpart.
-        """
+        data['link'] = self.link.to_spec() if self.link else None
```

Comment:
Should `to_serialized` be used here as well?

Reply:
Nope, it's correct, but lacking a comment. The idea is `Links` already has a built-in capability of "from spec" conversion, it's used to create `Links` from regular specs, therefore we can use its `to_spec()` and pair it with `tmt.base.Links(serialized['link'])` in `from_serialized`.

I'll add a comment, or maybe I'll add a custom `to/from_serialized` to `Links` class, which would just proxy `to/from_spec` - suddenly, this seems like a better solution, sticking to serialization in serialization chain of methods.

Reply:
Although that would require yet another base class... I'll just add a comment.

Reply:
Comment added.